components/admin_console/billing/billing_subscriptions/limits.tsx

```diff
@@ -272,7 +309,111 @@ const Limits = (): JSX.Element | null => {
 
                 )}
             </div>
-            <TempLaunchModalsComponent/>
+        );
+    } else if (subscriptionProduct.sku === CloudProducts.PROFESSIONAL) {
+        title = (
+            <FormattedMessage
+                id='workspace_limits.upgrade_professional'
+                defaultMessage='Professional plan data limit'
+            />
+        );
+        description = intl.formatMessage(
+            {
+                id: 'workspace_limits.upgrade_reasons.professional',
+                defaultMessage: 'On the Professional plan you can only view the most recent {fileStorage} of files. With Enterprise, experience unlimited file storage.',
+            },
+            {
+                fileStorage: asGBString(cloudLimits?.files?.total_storage || FileSizes.Gigabyte * 250, intl.formatNumber),
+            },
+        );
+        currentUsage = cloudLimits?.files?.total_storage && (
+            <SingleInlineLimit
+                name={(
+                    <FormattedMessage
+                        id='workspace_limits.file_storage'
+                        defaultMessage='File Storage'
+                    />
+                )}
+                status={(
+                    <FormattedMessage
+                        id='workspace_limits.file_storage.single_usage'
+                        defaultMessage='{actual} of {limit}'
+                        values={{
+                            actual: asGBString(usage.files.totalStorage, intl.formatNumber),
+                            limit: asGBString(cloudLimits.files.total_storage, intl.formatNumber),
+
+                        }}
+                    />
+                )}
+                percent={Math.floor((usage.files.totalStorage / cloudLimits.files.total_storage) * 100)}
+                icon='icon-folder-outline'
+            />
+        );
+    }
+
+    let panelClassname = 'ProductLimitsPanel';
+    if (singleLimitPanel) {
+        if (props.showAnnualCard) {
+            panelClassname += ' ProductLimitsPanel--left-panel';
+        } else {
+            panelClassname += ' ProductLimitsPanel--single-panel';
+        }
+    }
+    let actionsClassname = 'ProductLimitsPanel__actions';
+    if (singleLimitPanel) {
+        actionsClassname += ' ProductLimitsPanel__actions--single';
+    }
+    return (
+        <div className={panelClassname}>
+            {title && <div className='ProductLimitsPanel__title'>
+                {title}
+            </div>}
+            {description && <div className='ProductLimitsPanel__description'>
+                {description}
+            </div>}
+            {currentUsage}
+            <div className={actionsClassname}>
+                {subscriptionProduct.sku === CloudProducts.STARTER && (
+                    <>
+                        <button
+                            onClick={() => {
```

Comment:
nit: it is easier to read and understand when the function is separated and named with a descriptive name and then the onclick calls that function. Also, there is a performance topic about inline functions, I don't think that the performance is a big deal, but is another thing to keep in mind to avoid inline/anonimus functions

Reply:
> easier to read

0/5 on readability in this case; it gives a descriptive name (if I choose a good name), but makes the reader jump to another place in the code to see a fairly small function implementation. Since this function is used in several components, I think its worth extracting to a hook for consistency sake, and then by doing that it would have the descriptive name. Another PR introduces a hook for this, so I'm opting to defer doing this until then.

> performance ... inline functions ...

Agreed its probably not a meaningful difference either way in this case. It's not a hot render path for the application as a whole, or probably even for the subscriptions page.

A function defined as a function or a variable would have the same object identity inequality issue as an inline function unless you did useCallback or useMemo.

Doing useMemor or useCallback doesn't necessarily [increase performance](https://kentcdodds.com/blog/usememo-and-usecallback); there's several more memory allocations, especially for the tracking between renders. And the inline function is only declared conditionally, so in teh case the user's instance is not on the cloud starter plan, less work would be done because the inline function wouldn't be declared at all, but would need to be if we're using useCallback/useMemo.